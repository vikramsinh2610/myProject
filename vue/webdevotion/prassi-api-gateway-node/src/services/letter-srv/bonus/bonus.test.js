const Bonus = require('./bonus');

describe('Bonus', () => {
  test('should have these properties', () => {
    const bonus = new Bonus({
      amount: 1000,
      type: 'guaranteed',
      paymentTime: 'prepayment',
    });
    expect(bonus).toHaveProperty('amount');
    expect(bonus).toHaveProperty('type');
    expect(bonus).toHaveProperty('paymentTime');
  });

  test('should return it s JSON Schema', () => {
    expect(Bonus.getJSONSchema()).toMatchSnapshot();
  });
});
