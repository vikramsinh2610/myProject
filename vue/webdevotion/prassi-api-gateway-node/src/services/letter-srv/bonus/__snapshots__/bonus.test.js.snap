// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bonus should return it s JSON Schema 1`] = `
Object {
  "description": "Bonus",
  "properties": Object {
    "amount": Object {
      "type": "integer",
    },
    "paymentTime": Object {
      "enum": Array [
        "prepayment",
        "payment-delayed",
      ],
      "type": "string",
    },
    "type": Object {
      "enum": Array [
        "guaranteed",
        "conditioned",
      ],
      "type": "string",
    },
  },
  "required": Array [
    "amount",
    "type",
    "paymentTime",
  ],
  "type": "object",
}
`;
